### PR TITLE
Improve Repository Root Excel Export

### PR DESCRIPTION
**Pull request by Abdu-moustafa**
Improves the repository root Excel export feature by adding the folder UID and relative path to each entry
For [TI-131](https://4teamwork.atlassian.net/browse/TI-131)

**before**:

before

**After**:
after

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- Upgrade-Steps:
  - [ ] SQL Operations do not use imported model (see [docs](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/994344994/Upgrade-Steps))
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
  - DB-Schema migration
    - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
    - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- API change:
  - [ ] Documentation is updated
  - [ ] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- Bug fixed:
  - [ ] Resolved any Sentry issues caused by this bug
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Further improvements needed:
  - [ ] Create follow-up stories and link them in the PR and Jira issue
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

[TI-131]: https://4teamwork.atlassian.net/browse/TI-131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/repository/browser/excel_export.py [hunk 1]
```diff
@@ -228,6 +241,17 @@ def get_groupnames_with_local_or_inherited_role(self, rolename):
                 groups.append(role.get('id'))
         return groups
 
+    def get_folder_path(self):
+        """"Returns the relative path of the repository folder
+        Example:
+            If the physical path components are ('', 'fd', 'ordnungssystem')
+            this method returns '/ordnungssystem'.
+        """
+        return os.path.join("/", *self._repofolder.getPhysicalPath()[2:])
```
Comment (elioschmutz) [inline opengever/repository/browser/excel_export.py:250]: Use or overtake the `to_relative_path` function from [`opengever.core/opengever/base/solr/fields.py`](https://github.com/4teamwork/opengever.core/blob/master/opengever/base/solr/fields.py#L168)

`to_relative_path(self._repofolder.getPhysicalPath())`

File: opengever/repository/browser/excel_export.py [hunk 2]
```diff
@@ -141,8 +142,20 @@ def generate_report(request, context):
         u'label_groupnames_with_local_or_inherited_taskresponsible_role',
         default=u'Task responsible',
     ))
+
+    label_repositoryfolder_UID = repository_translater(_(
```
Comment (buchi) [inline opengever/repository/browser/excel_export.py:146]: naming should be lowercase with underscores.
```suggestion
    label_repositoryfolder_uid = repository_translater(_(
```

File: opengever/repository/browser/excel_export.py [hunk 3]
```diff
@@ -141,8 +142,20 @@ def generate_report(request, context):
...
+        default=u'UID',
+    ))
+    label_repositoryfolder_path = repository_translater(_(
+        u'label_repositoryfolder_path',
+        default=u'Path',
+    ))
+
     column_map = (
         {'id': 'get_repository_number', 'title': label_repository_number, 'fold_by_method': repository_number_to_outine_level, 'callable': True},  # noqa
+        {'id': 'get_folder_UID', 'title': label_repositoryfolder_UID, 'callable': True},
```
Comment (buchi) [inline opengever/repository/browser/excel_export.py:157]:
```suggestion
        {'id': 'get_folder_uid', 'title': label_repositoryfolder_uid, 'callable': True},
```

File: opengever/repository/browser/excel_export.py [hunk 4]
```diff
@@ -228,6 +241,17 @@ def get_groupnames_with_local_or_inherited_role(self, rolename):
...
 
+    def get_folder_path(self):
+        """"Returns the relative path of the repository folder
+        Example:
+            If the physical path components are ('', 'fd', 'ordnungssystem')
+            this method returns 'ordnungssystem'.
+        """
+        return to_relative_path('/'.join(self._repofolder.getPhysicalPath()))
+
+    def get_folder_UID(self):
```
Comment (buchi) [inline opengever/repository/browser/excel_export.py:252]:
```suggestion
    def get_folder_uid(self):
```

**Review by buchi — CHANGES_REQUESTED** [review]
Just some objections regarding naming. Maybe have a look at https://peps.python.org/pep-0008/#naming-conventions

**Review by buchi — APPROVED** [review]

**Pull request merged**